docs/spec/distributing.rst

```diff
@@ -10,72 +10,161 @@ Stub files
 
 (Originally specified in :pep:`484`.)
 
-Stub files are files containing type hints that are only for use by
-the type checker, not at runtime.  There are several use cases for
-stub files:
+*Stub files*, also called *type stubs*, provide type information for untyped
+Python packages and modules. Stub files serve multiple purposes:
+
+* They are the only way to add type information to extension modules.
+* They can provide type information for packages that do not wish to
+  add them inline.
+* They can be distributed separately from the package or module that they
+  provide types for. The latter is referred to as the *implementation*.
+  This allows stubs to be developed at a different pace or by different
+  authors, which is especially useful when adding type annotations to
+  existing packages.
+* They can act as documentation, succinctly explaining the external
+  API of a package, without including implementation details or private
+  members.
+
+Stub files use a subset of the constructs used in Python source files, as
+described in :ref:`stub-file-supported-constructs` below. Type checkers should
+parse a stub that uses only such constructs without error and not interpret any
+construct in a manner contradictory to this specification. However, type
+checkers are not required to implement checks for all of these constructs and
+can elect to ignore unsupported ones. Additionally, type checkers can support
+constructs not described here.
+
+If a stub file is found for a module, the type checker should not read the
+corresponding "real" module. See :ref:`mro` for more information.
+
+.. _stub-file-syntax:
+
+Syntax
+^^^^^^
+
+Stub files are syntactically valid Python files in the earliest Python version
+that is not yet end-of-life. They use a ``.pyi`` suffix. The Python syntax used
+by stub files is independent from the Python versions supported by the
+implementation, and from the Python version the type checker runs under (if
+any). Therefore, stubs may use the latest syntax features available in
+the earliest supported version, even if the implementation supports older
+versions.
+
+For example, Python 3.7 added the ``async`` keyword (see :pep:`492`). Stubs may
+use it to mark coroutines, even if the implementation still uses the
+``@coroutine`` decorator. On the other hand, the ``type`` soft keyword from
+:pep:`695`, introduced in Python 3.12, is not valid in stubs until Python 3.11
+reaches end-of-life in October 2027.
+
+Stubs are treated as if ``from __future__ import annotations`` is enabled. In
+particular, forward references do not need to be quoted, and syntax from newer
+versions than otherwise supported may be used in annotation contexts. For
```

Comment:
Actually, I think this sentence does need to describe where syntax from newer versions can be used; otherwise, it just sounds like it's directly contradicting the earlier info about what syntax stubs use. I changed this to say "in annotation expressions" rather than "in annotation contexts." Hopefully that's clearer, since "annotation expression" is a term that is defined in the glossary.